InternetSemLimites/api/views.py

```diff
@@ -49,8 +49,21 @@ def provider_detail(request, pk):
 
 
 @csrf_exempt
-def provider_new(request):
+def provider(request, pk):
+    routes = {
+        'GET': provider_detail,
+        'POST': provider_edit
+    }
+
+    route = routes.get(request.method)
+
+    if route:
+        return route(request, pk)
+    return HttpResponseNotAllowed(['GET', 'POST'])
```

Comment:
Tu é muito Pythonico meu brother hahah Dá pra fazer com `HttpResponseNotAllowed([method for method in routes])`.